### PR TITLE
[REEF-1333] Type cast warnings in our C++ code

### PR DESCRIPTION
**Pull request by anicoara**
This change:
  * Calls the `JNIENV_` methods that match the method signatures of
`class InteropReturnInfo`.
  * Removes some function calls that were NOOPs.

JIRA:
  [REEF-1333](https://issues.apache.org/jira/browse/REEF-1333)

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Bridge/InteropReturnInfo.cpp [hunk 1]
```diff
@@ -64,24 +64,20 @@ namespace Org {
...
-					_env->CallObjectMethod(_jobjectInteropReturnInfo, _jmidAddExceptionString, ret);
-					HasExceptions();
+					_env->CallVoidMethod(_jobjectInteropReturnInfo, _jmidAddExceptionString, ret);
 				}
 
 				Boolean InteropReturnInfo::HasExceptions() {
-					jobject obj = _env->CallObjectMethod(_jobjectInteropReturnInfo, _jmidHasExceptions);
-					return obj != NULL;
+					jboolean obj = _env->CallBooleanMethod(_jobjectInteropReturnInfo, _jmidHasExceptions);
+					return obj != 0;
```
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Bridge/InteropReturnInfo.cpp]: `JNI_FALSE`

**Comment by afchung** [comment]
Was in the middle of the review when I realized it doesn't seem like this class is used at all in the codebase. I'm not entirely familiar with this code, @beysims could you please comment? Alternatively, @anicoara could you please remove code related with `InteropReturnInfo` and see if C# tests still work? (Namely `InteropReturnInfo.cpp`, `IInteropReturnInfo.cs`, and `InteropReturnInfo.h`. I can run the C# tests on Hadoop if you need help. Thanks!

**Comment by afchung** [comment]
@anicoara I've also left a comment on the linked JIRA item, please have a look. Thanks!

**Comment by anicoara** [comment]
@afchung I couldn't find any references either, through just basic grep. I removed the files, and ran the local tests -- I did have two failures for `msbuild .\lang\cs\TestRunner.proj` but they look unrelated to me.
Please go ahead and try the C# tests on Hadoop as I don't have it setup yet. Do you want me to update the commit with the files removed to make it easier?

**Comment by anicoara** [comment]
@afchung I've added the commit that deletes the files. Feel free to comment on which commit should be kept, and I'll update the PR afterwards, removing the unnecessary one.

**Comment by jwang98052** [comment]
I will do a clean build and test.

**Comment by afchung** [comment]
@jwang98052 have you run the tests? If not, I can do it. Thanks!

**Comment by jwang98052** [comment]
@afchung Please go ahead. I couldn't get this branch.

**Comment by afchung** [comment]
@anicoara this is merged, but unfortunately I've referenced the wrong PR number so this PR was not automatically closed. Could you please close this for me? Thanks and sorry for the trouble.

**Pull request closed without merging**